cylc/flow/pathutil.py

```diff
@@ -28,10 +28,20 @@
 )
 from cylc.flow.platforms import get_localhost_install_target
 
+
 # Note: do not import this elsewhere, as it might bypass unit test
 # monkeypatching:
 _CYLC_RUN_DIR = os.path.join('$HOME', 'cylc-run')
 
+EXPLICIT_RELATIVE_PATH_REGEX = re.compile(
+    rf'''
+    ^({re.escape(os.curdir)}|{re.escape(os.pardir)})
+    ({re.escape(os.sep)}|$)
+    ''',
+    re.VERBOSE
+)
+"""Matches relative paths that are explicit (starts with ./)"""
```

Comment:
Just curious, is this path format officially called "explicit relative"? If you invented the term, I quite like it :+1:  Although arguably it's really an absolute path.

Reply:
I can't seem to find any terminology for relative paths for differentiating between whether they start with `./` or not. Explicit and implicit seems to be something one of us (maybe Tim, as author of the issue #4676) came up with.

Maybe we should avoid calling relative paths that don't start with `./` as relative paths at all; maybe calling them workflow source names would be less confusing? The arg could then be `SOURCE_NAME | PATH` where `PATH` can be relative or absolute but must start with `./` if relative.

Reply:
The trouble is that conflicts with standard unix terminology, where a relative path doesn't have to start with `./`.  

IMO `./blah` and `../blah` should really be considered absolute paths, where `.` is short for `$PWD` which expands to an absolute path. Like `~/blah` is an absolute path even though it is relative to `$HOME`.  But, that doesn't seem to be the prevailing opinion out there.

Reply:
> If you invented the term

MB but I think it's valid and helpful here.

> The trouble is that conflicts with standard unix terminology
> IMO ./blah and ../blah should really be considered absolute paths

I don't think that's correct, technically `.` and `..` *are* relative paths implemented at the FS level and visible in FS listings. So `.` is *not* short for `$PWD` though it happens to *resolve* to the same thing.

* `$PWD/foo` - the `$PWD` is *expanded* by the shell.
* `./foo` - the `./` is *resolved* by the FS.

Reply:
Technically true, but I doubt that matters to users.

BTW I didn't mean that `.` is *literally* short for `$PWD` and, like the shell variable, expanded by the shell ... just that it is short for the present working directory.  A true relative path is implicit. Presumably the main reason for the existence of `..` is to make changing up a level easier, and `.`, to force the behaviour of programs that don't interpret (implicit) relative paths properly.  So I'd still argue that `rel/path` is a proper relative path, but `./rel/path` is kind of an absolute path masquerading as a relative path.

Reply:
Explicit relative and absolute paths are handled the same by the CLI so this disparity is not user facing. This is internal technical documentation in a docstring.

Reply:
(Which still needs to be understood at a glance by future developers).